### ISSUE TITLE
Multi-threading doesn't seem to work

### ISSUE THREAD

**Issue by julhe** [aclysma/basis-universal-rs#12]
When comparing the performance against the original [CLI / basisu.exe](https://github.com/BinomialLLC/basis_universal) version, it appears that multithreading doesn't work. 

The timings in get on my machine with a test image:

- CLI (no flags): 3.111 secs
- CLI (-no_multithreading flag): 13.649 secs
- Rust (see code below): 12.135 secs

I'm on Windows 10 x64 with this toolchain:
```
stable-x86_64-pc-windows-msvc (default)
rustc 1.68.0 (2c8cc3432 2023-03-06)
````

My code for reference
```rust
use basis_universal::*;
let img = image::open("test_image.png").unwrap();
basis_universal::encoder_init();

let mut params = CompressorParams::new();
params.set_print_status_to_stdout(true);

let mut compressor_image = params.source_image_mut(0);
compressor_image.init(&img.to_rgba8(), img.width(), img.height(), 4);

let mut compressor = encoding::Compressor::new(24); //explicit multi-threading
let compression_time = unsafe {
    compressor.init(&params);
    let t0 = std::time::Instant::now();
    compressor.process().unwrap();
    let t1 = std::time::Instant::now();
    t1 - t0
};

println!("compression took {}secs", compression_time.as_secs_f32());
```

**Comment by aclysma** [comment]
I took a quick look and it appears the thread count is passed directly to the upstream basis library.

https://github.com/aclysma/basis-universal-rs/blob/2ae4264aa46d0f2d5f92a40046d40851f19b14db/basis-universal/src/encoding/compressor.rs#L45-L51

https://github.com/aclysma/basis-universal-rs/blob/2ae4264aa46d0f2d5f92a40046d40851f19b14db/basis-universal-sys/src/encoding_bindings.rs#L599-L601

Doesn't seem like there's much to go wrong here unless there is some other configuration param that needs to be set (like "enabling" MT) or if MT compression is only supported for particular formats, size of textures, quality settings, etc.

I would consider trying to reproduce this with their library directly (maybe hack some sample code in their repo if you can) or trace it through with a debugger to see where that thread count goes and confirm that it's being used and is set to the expected value.

**Comment by julhe** [comment]
Mistery solved! Thread count turned out to be fine on the C++ side. Enconding parameters where also normal. Even enabling SEE didn't matched the C++ version. At point I noticed that the debug logs I got from the Rust version where slightly different from a C++ equvialent I build. 

Turns out that this crate simply binds to Basis Universal 1.15, while the most recent one apparently had some hefty performance improvements. Updating to 1.16 brings the Rust version on-pair with the C++ one. If you want I can make a PR for this.
